consensus/consensus-types/src/proposal_msg.rs

```diff
@@ -54,18 +54,23 @@ impl ProposalMsg {
             self.sync_info.highest_quorum_cert().certified_block().id(),
             self.proposal.parent_id(),
         );
-        let previous_round = self.proposal.round() - 1;
+        let previous_round = self.proposal.round().overflowing_sub(1);
+        ensure!(
+            !previous_round.1,
+            "Encountered integer overflow when creating proposal {}",
+            self.proposal
+        );
```

Comment:
I think this is clearer:

```rust
let previous_round = self.proposal.round().checked_sub(1).ok_or_else(|| anyhow!("proposal round overflowed!"))?;
```

(following @JoshLind 's style for overflow errors)

Reply:
@mimoo  fixed ! 👍